internal/status/route_status.go

```diff
@@ -48,17 +47,41 @@ func ParseOptionsFromMetadata(options []UpdateOptions, bulk bool) ([]string, []U
 
 				ingress := ingressArr[0] + "/" + ingressArr[1]
 				option.IngSvc = ingress
-				objectsToUpdate = append(objectsToUpdate, ingress)
-				updateIngressOptions = append(updateIngressOptions, option)
+				if opt, ok := updateIngressOptions[ingress]; ok {
+					for _, hostname := range option.ServiceMetadata.HostNames {
+						if !utils.HasElem(opt.ServiceMetadata.HostNames, hostname) {
+							opt.ServiceMetadata.HostNames = append(opt.ServiceMetadata.HostNames, option.ServiceMetadata.HostNames...)
+							updateIngressOptions[ingress] = opt
+						}
+					}
+				} else {
+					updateIngressOptions[ingress] = option
+				}
 			}
 		} else {
+			// insecure VSes, servicemetadata comes from Pools.
 			ingress := option.ServiceMetadata.Namespace + "/" + option.ServiceMetadata.IngressName
 			option.IngSvc = ingress
-			objectsToUpdate = append(objectsToUpdate, ingress)
-			updateIngressOptions = append(updateIngressOptions, option)
+			if opt, ok := updateIngressOptions[ingress]; ok {
+				for _, hostname := range option.ServiceMetadata.HostNames {
+					if !utils.HasElem(opt.ServiceMetadata.HostNames, hostname) {
+						opt.ServiceMetadata.HostNames = append(opt.ServiceMetadata.HostNames, option.ServiceMetadata.HostNames...)
+						updateIngressOptions[ingress] = opt
+					}
+				}
+			} else {
+				updateIngressOptions[ingress] = option
+			}
 		}
 	}
-	return objectsToUpdate, updateIngressOptions
+
+	ingressesToUpdate := make([]string, len(updateIngressOptions))
+	i := 0
+	for k := range updateIngressOptions {
+		ingressesToUpdate[i] = k
```

Comment:
An append would also work here, and we would not need the index i. This is also fine though.